resource/environment/integration.js

```diff
@@ -329,11 +168,14 @@ const upgradeStack = Promise.coroutine(function* ( slack, dockerhub, github, env
 			}
 		};
 	}
+	
 	//get the environments
 	const storedEnvironments = yield environment.getAll().catch( () => undefined );
 	if ( storedEnvironments === undefined ) { 
-		return readError;
+		return {status: 404, data: {message: "Unable to get information from the database"}}; 
```

Comment:
HTTP code here should indicate a server level error 5xx. In this endpoint a 404 would be appropriate if the stack being requested to upgrade did not exists or something similar (such as information not being in github)